galasa-parent/dev.galasa.framework.api.cps/src/main/java/dev/galasa/framework/api/cps/internal/common/GalasaProperty.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.framework.api.cps.internal.common;
+
+import java.util.Map;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+
+import dev.galasa.framework.spi.utils.GalasaGsonBuilder;
+
+public class GalasaProperty {
+    static final Gson gson = GalasaGsonBuilder.build();
+    
+    public String apiVersion = "galasa-dev/v1alpha1";
+    public final String kind = "GalasaProperty"; 
+    public GalasaPropertyMetadata metadata ;
+    public GalasaPropertyData data;
```

Comment:
I would change these to private and make getters to avoid directly referencing them.